### ISSUE TITLE
U-Net 3+ implementation

### ISSUE THREAD

**Issue by AKMourato** [PaddlePaddle/PaddleSeg#1432]
Greetings chaps,

Following the official implementation of U-Net 3+:
- https://github.com/ZJUGiveLab/UNet-Version
- https://arxiv.org/abs/2004.08790

In the deep supervision module is applied a hybrid loss that is the sum between the focal loss, the MS-SSIM loss, and the IoU loss. However, in your implementation I don't see it defined in the same manner.
Could you clarify the modifications that you have employed here?

Thanks

Ps: (don't forget to add the weights loading argument, thx)

**Comment by wuyefeilin** [comment]
@geoyee The user has a question when using U-Net 3+. Can you help answer it?

**Comment by geoyee** [comment]
@wuyefeilin @AKMourato Sorry, this is only the network code (UNet3+) of the [repo](https://github.com/ZJUGiveLab/UNet-Version), and the loss code is **not added**. I will improve this in the near future.

**Comment by wuyefeilin** [comment]
> @wuyefeilin @AKMourato Sorry, this is only the network code (UNet3+) of the [repo](https://github.com/ZJUGiveLab/UNet-Version), and the loss code is **not added**. I will improve this in the near future.

Good, looking forward to your improvement.

**Comment by geoyee** [comment]
@AKMourato Hello, These losses have been perfected [here](https://github.com/Nanyi-Brigade/PaddleSeg/tree/develop), but I can't submit to PaddleSeg, You can refer to the above repo.
![image](https://user-images.githubusercontent.com/71769312/137835573-089b92d8-9435-4ff7-90c7-da0bb80b2851.png)
## Reasons
1. In binary classification, PaddleSeg's `num_classes=2` and used `argmax`, but in [https://github.com/ZJUGiveLab/UNet-Version](https://github.com/ZJUGiveLab/UNet-Version), `num_classes=1` and used `threshold`. This is **incompatible**(evaluation indicators and prediction errors).
2. In this [repo](https://github.com/avBuffer/UNet3plus_pth/blob/263534e4a48964e907324622b14b90f1c3b4270d/train.py), training codes are not used IOU Loss and MS-SSIM Loss. When I use this in a dog segmentation task, the effect is not very good.
3. Sorry, I didn't find the weights.

*THX, if there are any questions and ideas, I hope to communicate.*
